presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -29,18 +30,32 @@
 
 public class PlanNodeSearcher
 {
+    @Deprecated
```

Comment:
@kokosing I think of traits as an extension to the iterative optimizer. We should be aiming to enable iterative optimizer as the default one, then we can refactor/revamp it's traversal strategy with traits etc. The sooner we enable iterative optimizer as default the sooner it becomes robust.
Therefore, stalling the process of porting of some rules to the moment when we have traits, is worse strategy then porting them and refactoring later, when new features to the optimizer are in. I think baby steps are always better.